src/miral/static_display_config.cpp

```diff
@@ -0,0 +1,376 @@
+/*
+ * Copyright © 2018 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Alan Griffiths <alan@octopull.co.uk>
+ */
+
+#include "static_display_config.h"
+
+#include <mir_toolkit/mir_client_library.h>
+
+#include <mir/log.h>
+
+
+#include "yaml-cpp/yaml.h"
+
+#include <algorithm>
+#include <fstream>
+#include <sstream>
+#include <iostream>
+
+namespace mg = mir::graphics;
+using namespace mir::geometry;
+
+namespace
+{
+constexpr char const* const output_id = "output_id";
```

Comment:
One too many const? constexpr makes output_id a const pointer, which points to a "const char". I suspect that last "const" is unnecessary